### PR TITLE
Update version of clustalo

### PR DESCRIPTION
**Pull request by souravsingh**
* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by cbrueffer** [comment]
I looked into this a while ago, there's no updated OSX binary available. Only updating the Linux one as currently done won't work. Maybe ask the Clustal folks to update the OSX binary as well?

**Comment by souravsingh** [comment]
I just noticed this. I believe this could be fixed by creating separate recipes for Linux and OSX.

**Comment by cbrueffer** [comment]
That may work, but I don't think it's a good solution. Way easier to send an email :-)

**Comment by souravsingh** [comment]
Ok, I have sent a mail to the maintainers of Clustalo

**Comment by cbrueffer** [comment]
Great, thanks!

**Comment by eseiler** [comment]
Hey,

I have two suggestions:

Either
(1) Use different versions for Linux and OSX:
```
{% set version = "1.2.3" %} # [osx] 
{% set version = "1.2.4" %} # [linux]
```

Or
(2) Build from source:
meta.yaml

```
{% set version = "1.2.4" %}

package:
  name: "clustalo"
  version: "{{ version }}"

source:
  url: http://www.clustal.org/omega/clustal-omega-{{ version }}.tar.gz # [not win]
  sha256: 8683d2286d663a46412c12a0c789e755e7fd77088fb3bc0342bb71667f05a3ee # [not win]

requirements:
  build:
    - gcc
    - argtable2
  run:
    - libgcc
    - argtable2

build:
  number: 0

test:
  commands:
    - "clustalo --help > /dev/null"

about:
  home: "http://www.clustal.org/omega/"
  license: "GPLv2"
  summary: "Latest version of Clustal: a multiple sequence alignment program for DNA or proteins"
```

build.sh
```
#!/bin/bash

./configure --prefix=$SRC_DIR
make
make install

mkdir -p $PREFIX/bin
cp $SRC_DIR/bin/clustal* $PREFIX/bin/$PKG_NAME
chmod +x $PREFIX/bin/$PKG_NAME
```

It's usually better to build it from source if possible and it would circumvent the version problem in the future.
I tested both (1) and (2) on travis and they worked just fine.

There may be some additional options you want to pass to ./configure (see their install manual), but this is all that is needed to be compiled from source.

File: recipes/clustalo/build.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+./configure --prefix=$SRC_DIR
+make
+make install
```
Comment (cbrueffer) [inline recipes/clustalo/build.sh]: I think this shouldn't be necessary given that the binary is copied by hand later on?

**Comment by eseiler** [comment]
Yes, this may be a good idea.
If it just copies the binaries, it shouldn't matter much, though.

File: recipes/clustalo/meta.yaml [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
-  md5: d6f953f498ea4fa1d08f57ea8cd731d0 # [linux]
-  fn: clustal-omega-{{ version }}-macosx # [osx]
-  url: http://www.clustal.org/omega/clustal-omega-{{ version }}-macosx # [osx]
-  md5: e8ec7292c258b8b1ef4f3780b3952fe8 # [osx]
+  url: http://www.clustal.org/omega/clustal-omega-{{ version }}.tar.gz # [not win]
+  sha256: 8683d2286d663a46412c12a0c789e755e7fd77088fb3bc0342bb71667f05a3ee # [not win]
+
+requirements:
+  build:
+    - gcc
```
Comment (bgruening) [inline recipes/clustalo/meta.yaml]: Can you please try to use llvm for osx as written in our documentation: https://bioconda.github.io/guidelines.html#c-c

File: recipes/clustalo/meta.yaml [hunk 3]
```diff
@@ -1,25 +1,35 @@
...
   version: "{{ version }}"
 
 source:
-  fn: clustalo-{{ version }}-Ubuntu-x86_64 # [linux]
-  url: http://www.clustal.org/omega/clustalo-{{ version }}-Ubuntu-x86_64 # [linux]
-  md5: d6f953f498ea4fa1d08f57ea8cd731d0 # [linux]
-  fn: clustal-omega-{{ version }}-macosx # [osx]
-  url: http://www.clustal.org/omega/clustal-omega-{{ version }}-macosx # [osx]
-  md5: e8ec7292c258b8b1ef4f3780b3952fe8 # [osx]
+  url: http://www.clustal.org/omega/clustal-omega-{{ version }}.tar.gz # [not win]
```
Comment (bgruening) [inline recipes/clustalo/meta.yaml]: you don't need the [not win] selector, we don't compile for Windows

File: recipes/clustalo/meta.yaml [hunk 4]
```diff
@@ -1,25 +1,35 @@
...
+
+requirements:
+  build:
+    - gcc # [linux]
+    - llvm # [osx]
+    - libgcc # [linux]
+    - argtable2
+  run:
+    - llvm # [osx]
+    - gcc # [linux]
```
Comment (bgruening) [inline recipes/clustalo/meta.yaml]: gcc should not be a runtime dependency

File: recipes/clustalo/meta.yaml [hunk 5]
```diff
@@ -1,25 +1,35 @@
...
-  url: http://www.clustal.org/omega/clustal-omega-{{ version }}-macosx # [osx]
-  md5: e8ec7292c258b8b1ef4f3780b3952fe8 # [osx]
+  url: http://www.clustal.org/omega/clustal-omega-{{ version }}.tar.gz # [not win]
+  sha256: 8683d2286d663a46412c12a0c789e755e7fd77088fb3bc0342bb71667f05a3ee # [not win]
+
+requirements:
+  build:
+    - gcc # [linux]
+    - llvm # [osx]
+    - libgcc # [linux]
```
Comment (bgruening) [inline recipes/clustalo/meta.yaml]: libgcc should not be a compile time dependency

**Comment by souravsingh** [comment]
Looks good now. @bgruening Would you like to take a look before merging?

**Pull request merged**